community/cypher/cypher-compiler-2.3/src/main/scala/org/neo4j/cypher/internal/compiler/v2_3/pipes/QueryState.scala

```diff
@@ -19,25 +19,26 @@
  */
 package org.neo4j.cypher.internal.compiler.v2_3.pipes
 
-import java.util.UUID
+import java.util.{Set, UUID}
 
 import org.neo4j.cypher.internal.compiler.v2_3._
 import org.neo4j.cypher.internal.compiler.v2_3.commands.expressions.PathValueBuilder
 import org.neo4j.cypher.internal.compiler.v2_3.spi.QueryContext
 
+import scala.collection.mutable
+
 case class QueryState(query: QueryContext,
                       resources: ExternalResource,
                       params: Map[String, Any],
                       decorator: PipeDecorator,
                       timeReader: TimeReader = new TimeReader,
                       var initialContext: Option[ExecutionContext] = None,
-                      queryId: AnyRef = UUID.randomUUID().toString) {
-
+                      queryId: AnyRef = UUID.randomUUID().toString,
+                      triadicSets: mutable.Map[String, Set[Long]] = new mutable.HashMap[String, Set[Long]]()) {
```

Comment:
Scala collections show up way too much in profiling, so I wanted to avoid adding more of it.